elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -263,12 +266,382 @@ private Object fetchRelationship(Environment context) {
+
+                    /* fetch id value fed to current argument */
+                    Optional<String> relationshipId = relationshipAttributes.entrySet().stream()
+                            .filter(entry -> idFieldName.equalsIgnoreCase(entry.getKey()))
+                            .map(e -> (String)e.getValue())
+                            .findFirst();
+
+                    /* create new resources */
+                    relationshipResource.add(updateOrCreateAttributes(relationshipAttributes, relationshipId, requestScope,
+                            resourceClass, outputType, relationship.getKey(), newParent));
+
+                    /* fetch any further nested relationships for later processing */
+                    Map<String, Object> nestedRelationships = stripAttributes(new ArrayList<>(Arrays.asList(entries)), resourceClass,
+                            dictionary);
+
+                    /* loop through these nested relationships and add them one by one on the queue */
+                    for(Map.Entry<String, Object> nestedRelation : nestedRelationships.entrySet()) {
+                        toVisit.add(new Triplet((List<Map<String, Object>>) nestedRelation.getValue(),
+                                relationshipResource.iterator().next(), nestedRelation.getKey()));
+                    }
+                }
+
+                /* update the relationship between parent and newly created/updated nested resources */
+                newParent.updateRelation(relationship.getKey(), relationshipResource);
+                returnOps.addAll(relationshipResource);
+            }
+        }
+        return returnOps;
+    }
+
+    /**
+     * updates or creates existing/new entities
+     * @param input data input
+     * @param id id
+     * @param requestScope request scope
+     * @param outputType graphql output type
+     * @param fieldName field name
+     * @param parent parent resource
+     * @return {@link PersistentResource} object
+     */
+    private PersistentResource updateOrCreateAttributes(Map<String, Object> input, Optional<String> id,
+                                              RequestScope requestScope, Class<?> loadClass,
+                                              GraphQLType outputType, String fieldName,
+                                              PersistentResource parent) {
+        if(!id.isPresent()) { /* no id(s) provided, must create new */
+            return createObject(input, fieldName, requestScope, Optional.empty(), outputType, parent);
+        }
+
+        Set<PersistentResource> fetchEntries;
+        try {
+            if(parent == null) {
+                fetchEntries = fetchObject(requestScope, loadClass,
+                        Optional.of(Arrays.asList(id.orElse(null))), Optional.empty(),
+                        Optional.empty(), Optional.empty(), Optional.empty());
+            } else {
+                fetchEntries = (Set<PersistentResource>) fetchRelationship(parent, fieldName);
```

Comment:
This seems odd to me.  If we are doing a fetch, we should be doing it by ID.  Not fetching the entire collection.  We should be calling a fetch function that only returns a single item (in both cases)